opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -911,10 +930,23 @@ namespace Opm {
             WellGroupHelpers::updateGuideRates(fieldGroup, schedule(), summaryState, this->phase_usage_, reportStepIdx, simulationTime,
                                         this->wellState(), this->groupState(), comm, &this->guideRate_, pot, local_deferredLogger);
         }
-        last_report_.converged = true;
-        last_report_.assemble_time_well += perfTimer.stop();
+
+        // Maybe do a recursive call to iterate network and well controls.
+        if (network_changed) {
+            if (shouldBalanceNetwork(reportStepIdx, iterationIdx)) {
+                const auto& balance = schedule()[reportStepIdx].network_balance();
+                // Iterate if not converged, and number of iterations is not yet max (NETBALAN item 3).
+                if (recursion_level <= balance.pressure_max_iter() && network_imbalance > balance.pressure_tolerance()) {
```

Comment:
Should here be something like 

```c++
recursion_level + 1 < balance.pressure_max_iter()
```

since recursion_level begins with `0`.

Reply:
Considering the corner case, balance.pressure_max_iter() == 1, I reasoned that we should do the recursive call once, which is what will happen here. But I think you may be right: If we count the number of times we calculate the network pressures, given some rates, and consider that to be the relevant "number of iterations", then we should reduce it by 1, making it < rather than <=. But adding a 1, skewing it one more step, would be wrong I think.

Reply:
At the location of the code, do you consider we already did one iteration/recursion? 

With the call above 
```c++
        const auto [network_changed, network_imbalance] = updateWellControls(local_deferredLogger, /* check group controls */ true);
```

Reply:
Yes, if we consider a single initial calculation of node pressures as an iteration, no otherwise. I think now that yes is the correct answer: we are after all potentially changing the node pressures, so it is not "0 iterations", which should correspond to doing no changes. I will change the <= to a <.